src/_posts/platform/app/2000-01-01-crash.md

```diff
@@ -1,69 +1,217 @@
+done by checking the metrics, analyzing the logs, or setting up automated
+anomaly detection. The goal is to identify and fix potential issues before they
+occur.
+
+Finally, we strongly encourage developers to test their code, to follow the
+best practices and to conduct Q/A testings in a dedicated environment before
+migrating to production.
+
+### Recovering from a Runtime Error
+
+When a Runtime Error occurs and leads to an application crash, the platform
+automatically takes some remedial measures to try to recover from it. The very
+first action consists in logging the crash event in your dashboard. Once done,
+the platform uses its restart policy, described hereafter:
+
+The platform maintains a crash counter for each running container. This crash
+counter is set to zero by default.
+
+When a container crashes for the first time (when its crash counter equals
+zero), the platform sets the crash counter to one and then tries to restart the
+container. This is done as soon as the failure is detected.
+
+After a successful restart, the container enters a *warmup* period which always
+lasts 10 minutes. From here, we distinguish two main cases:
+
+1. The container successfully goes past the *warmup* period. In this case, the
+   platform considers that the restart operation fixed the issue and thus
+   resets the container's crash counter to zero.
+
+2. The container crashes again during the *warmup* period. In this case, the
+   platform immediately increments the crash counter by one.
+   - If the crash counter value is less than 13, then the platform waits some
+     time before restarting the container again. The duration of the waiting
+     period is given by the following formula:
+     `duration (minutes) = (crash counter value - 2) x 5`
+   - It the crash counter value equals 2, 5 or 12, the platform sends a warning
+     e-mail to the application's collaborators.
+   - If the crash counter value reaches 13, which means the container keeps
+     crashing after 12 delayed restarts, the platform gives up. The container
+     is crashed and won't be restarted anymore, which mean your application
+     may now be totally unavailable.
+
+#### Fixing Runtime Errors
+
+Besides the measures taken by the platform, we strongly advise you to carefully
+investigate the logs of your crashing application as soon as possible. Once
+your fix is ready, push your updated code to Scalingo to trigger a new
+deployment, hopefully resolving the issue.
+
+## Getting Notified When a Crash Occurs
+
+By default, the platform only notifies the application's collaborator(s) when
+a recurrent crash occurs (i.e. when an early Runtime Error keeps occuring, or
+when a Timeout Error occurs).
+
+You can modify this behavior by tweaking your
+[Notifier's configuration]({% post_url platform/app/2000-01-01-notification %}).
+The *app_crashed*, *app_crashed_repeated* and the *app_deploy* events can be
```

Comment:
suggestion: I would rather use code block rather than italic:
```suggestion
The `app_crashed`, `app_crashed_repeated` and the `app_deploy` events can be
```